dataflow/run-inference/README.md

```diff
@@ -0,0 +1,132 @@
+# Streaming RunInference LLM
+
+[![Open in Cloud Shell](http://gstatic.com/cloudssh/images/open-btn.svg)](https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/python-docs-samples&page=editor&open_in_editor=dataflow/run-inference/README.md)
+
+This sample shows how to run a [Large Language Model (LLM)](https://en.wikipedia.org/wiki/Large_language_model) in a streaming Dataflow pipeline.
+
+## Before you begin
+
+> 💡 Make sure you have followed the
+[Dataflow setup instructions](../README.md).
+
+```sh
+# Make sure you have the following environment variables.
+export PROJECT="my-google-cloud-project-id"
+export BUCKET="my-cloud-storage-bucket"  # without the "gs://" prefix
+export LOCATION="us-central1"  # or your location of choice
+```
+
+Create the following Pub/Sub topics:
+* `messages`: For the input messages to send to the model.
+* `responses`: For the model's responses.
+
+```sh
+export MESSAGES_TOPIC="projects/$PROJECT/topics/messages"
+export RESPONSES_TOPIC="projects/$PROJECT/topics/responses"
+
+# Create the Pub/Sub topics.
+gcloud pubsub topics create $MESSAGES_TOPIC
+gcloud pubsub topics create $RESPONSES_TOPIC
+```
+
+## Loading the `state_dict`
+
+LLMs can be **very large** models. Make sure the VMs you choose have enough memory to load them.
+Larger models generally give better results, but require more memory and can be slower to run on CPUs.
+You can add GPUs if you need faster responses.
+
+| Model                                   | Parameters | Memory needed |
+|-----------------------------------------|-----------:|--------------:|
+| [`google/flan-t5-small`][flan-t5-small] | 80M        | > 320 MB      |
+| [`google/flan-t5-base`][flan-t5-base]   | 250M       | > 1 GB        |
+| [`google/flan-t5-large`][flan-t5-large] | 780M       | > 3.2 GB      |
+| [`google/flan-t5-xl`][flan-t5-xl]       | 3B         | > 12 GB       |
+| [`google/flan-t5-xxl`][flan-t5-xxl]     | 11B        | > 44 GB       |
+| [`google/flan-ul2`][flan-ul2]           | 20B        | > 80 GB       |
+
+> 💡 Larger models require larger VMs to run.
+
+When running in Dataflow, Apache Beam's [`RunInference`](https://beam.apache.org/documentation/transforms/python/elementwise/runinference/) expects the model's  [`state_dict`](https://pytorch.org/tutorials/recipes/recipes/what_is_state_dict.html) to reside in Cloud Storage.
+
+Since LLMs can be so large, we want to save our `state_dict` as `float16` instead of the default `float32`.
+This means each parameter uses 16 bits instead of 32 bits, making the `state_dict` be half the size, which minimizes the time to load the model.
```

Comment:
```suggestion
With this configuration, each parameter uses 16 bits instead of 32 bits, making the `state_dict` half the size, which minimizes the time needed to load the model.
```